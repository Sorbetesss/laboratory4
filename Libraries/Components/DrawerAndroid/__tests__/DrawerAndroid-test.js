/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @format
 * @emails oncall+react_native
 * @flow
 */

'use strict';

const React = require('React');
const AndroidDrawerLayout = require('../DrawerLayoutAndroid.android');
const View = require('View');

const render = require('../../../../jest/renderer');

jest.mock('UIManager', () => ({
  getViewManagerConfig: jest.fn(() => ({Constants: {}})),
}));

describe('<AndroidDrawerLayout />', () => {
  it('should render as <AndroidDrawerLayout> when mocked', () => {
    const instance = render.create(
      <AndroidDrawerLayout renderNavigationView={() => <View />} />,
    );
    expect(instance).toMatchSnapshot();
  });

  it('should shallow render as <DrawerLayoutAndroid> when mocked', () => {
    const output = render.shallow(
      <AndroidDrawerLayout renderNavigationView={() => <View />} />,
    );
    expect(output).toMatchSnapshot();
  });

  it('should shallow render as <DrawerLayoutAndroid> when not mocked', () => {
    jest.dontMock('DrawerLayoutAndroid');

    const output = render.shallow(
      <AndroidDrawerLayout renderNavigationView={() => <View />} />,
    );
    expect(output).toMatchSnapshot();
  });

  it('should render as <AndroidDrawerLayout> when not mocked', () => {
    jest.dontMock('DrawerLayoutAndroid');

    const instance = render.create(
      <AndroidDrawerLayout renderNavigationView={() => <View />} />,
    );
    expect(instance).toMatchSnapshot();
  });
});
