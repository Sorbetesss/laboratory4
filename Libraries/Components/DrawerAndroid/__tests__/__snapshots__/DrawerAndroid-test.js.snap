// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AndroidDrawerLayout /> should render as <AndroidDrawerLayout> when mocked 1`] = `
<AndroidDrawerLayout
  drawerBackgroundColor="white"
  onDrawerClose={[Function]}
  onDrawerOpen={[Function]}
  onDrawerSlide={[Function]}
  onDrawerStateChanged={[Function]}
  renderNavigationView={[Function]}
  style={
    Array [
      Object {
        "elevation": 16,
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    collapsable={false}
    style={
      Array [
        Object {
          "bottom": 0,
          "position": "absolute",
          "top": 0,
        },
        Object {
          "backgroundColor": "white",
          "width": undefined,
        },
      ]
    }
  >
    <View />
  </View>
</AndroidDrawerLayout>
`;

exports[`<AndroidDrawerLayout /> should render as <AndroidDrawerLayout> when not mocked 1`] = `
<AndroidDrawerLayout
  drawerBackgroundColor="white"
  onDrawerClose={[Function]}
  onDrawerOpen={[Function]}
  onDrawerSlide={[Function]}
  onDrawerStateChanged={[Function]}
  renderNavigationView={[Function]}
  style={
    Array [
      Object {
        "elevation": 16,
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    collapsable={false}
    style={
      Array [
        Object {
          "bottom": 0,
          "position": "absolute",
          "top": 0,
        },
        Object {
          "backgroundColor": "white",
          "width": undefined,
        },
      ]
    }
  >
    <View />
  </View>
</AndroidDrawerLayout>
`;

exports[`<AndroidDrawerLayout /> should shallow render as <DrawerLayoutAndroid> when mocked 1`] = `
<DrawerLayoutAndroid
  drawerBackgroundColor="white"
  renderNavigationView={[Function]}
/>
`;

exports[`<AndroidDrawerLayout /> should shallow render as <DrawerLayoutAndroid> when not mocked 1`] = `
<DrawerLayoutAndroid
  drawerBackgroundColor="white"
  renderNavigationView={[Function]}
/>
`;
